sqldelight-gradle-plugin/src/test/fixtures/nullability/expected/com/sample/Test1Model.java

```diff
@@ -0,0 +1,315 @@
+      + "SELECT test.*, test2._id, nullable_int, nonnull_int\n"
+      + "FROM test\n"
+      + "LEFT JOIN test2";
+
+  long _id();
+
+  @Nullable
+  String nullable_text();
+
+  @NonNull
+  String nonnull_text();
+
+  interface Join_tableModel<T1 extends Test1Model, T2 extends Test2Model> {
+    @NonNull
+    T1 test();
+
+    @NonNull
+    T2 test2();
+  }
+
+  interface Join_tableCreator<T1 extends Test1Model, T2 extends Test2Model, T extends Join_tableModel<T1, T2>> {
+    T create(@NonNull T1 test, @NonNull T2 test2);
+  }
+
+  final class Join_tableMapper<T1 extends Test1Model, T2 extends Test2Model, T extends Join_tableModel<T1, T2>> implements RowMapper<T> {
+    private final Join_tableCreator<T1, T2, T> creator;
+
+    private final Factory<T1> test1ModelFactory;
+
+    private final Test2Model.Factory<T2> test2ModelFactory;
+
+    Join_tableMapper(Join_tableCreator<T1, T2, T> creator, Factory<T1> test1ModelFactory, Test2Model.Factory<T2> test2ModelFactory) {
+      this.creator = creator;
+      this.test1ModelFactory = test1ModelFactory;
+      this.test2ModelFactory = test2ModelFactory;
+    }
+
+    @Override
+    @NonNull
+    public T map(@NonNull Cursor cursor) {
+      return creator.create(
+          test1ModelFactory.creator.create(
+              cursor.getLong(0),
+              cursor.isNull(1) ? null : cursor.getString(1),
+              cursor.getString(2)
+          ),
+          test2ModelFactory.creator.create(
+              cursor.getLong(3),
+              cursor.isNull(4) ? null : cursor.getLong(4),
+              cursor.getLong(5),
+              cursor.getLong(6)
+          )
+      );
+    }
+  }
+
+  interface Left_join_tableModel<T1 extends Test1Model, T2 extends Test2Model> {
+    @NonNull
+    T1 test();
+
+    @Nullable
+    T2 test2();
+  }
+
+  interface Left_join_tableCreator<T1 extends Test1Model, T2 extends Test2Model, T extends Left_join_tableModel<T1, T2>> {
+    T create(@NonNull T1 test, @Nullable T2 test2);
+  }
+
+  final class Left_join_tableMapper<T1 extends Test1Model, T2 extends Test2Model, T extends Left_join_tableModel<T1, T2>> implements RowMapper<T> {
+    private final Left_join_tableCreator<T1, T2, T> creator;
+
+    private final Factory<T1> test1ModelFactory;
+
+    private final Test2Model.Factory<T2> test2ModelFactory;
+
+    Left_join_tableMapper(Left_join_tableCreator<T1, T2, T> creator, Factory<T1> test1ModelFactory, Test2Model.Factory<T2> test2ModelFactory) {
+      this.creator = creator;
+      this.test1ModelFactory = test1ModelFactory;
+      this.test2ModelFactory = test2ModelFactory;
+    }
+
+    @Override
+    @NonNull
+    public T map(@NonNull Cursor cursor) {
+      return creator.create(
+          test1ModelFactory.creator.create(
+              cursor.getLong(0),
+              cursor.isNull(1) ? null : cursor.getString(1),
+              cursor.getString(2)
+          ),
+          cursor.isNull(3) && cursor.isNull(4) && cursor.isNull(5) && cursor.isNull(6)
```

Comment:
Future optimization: you only need to check a single non-nullable column in the joined table (likely the primary key)